### PR TITLE
Added Field Invoice Code To Company Model

### PR DESCRIPTION
**Pull request by tejaszambre**
- Added necessary validations on invoice code field
- Modified the ability of employees having role Finance, to have full access to Company
- Modified the UI for the new field added
- Added field active in company model
- Added a after save callback to update companys all project and user_projects if company if inactivated
- Removed the company destory action, path and all links related to delete functionality of company
- Removed the delete link of project from companys show template
- Modified templates of company for the new field added
- Modified the testcases for the same

### CODE REVIEW

**Comment by tejaszambre** [comment]
[Access of Company to Finance role](https://trello.com/c/JlAhBXHa/43-access-of-company-to-finance-role)

File: app/models/project.rb [hunk 1]
```diff
@@ -390,6 +391,19 @@ def update_user_projects
     user_projects.each { |user_project| user_project.update_attributes(end_date: end_date) }
   end
 
+  def update_team_members_end_date
+    team_user_projects = user_projects.where('$or': [
+      {:end_date.gte => end_date},
+      {end_date: end_date_was},
+      {end_date: nil}
```
Comment (Ashish944) [inline app/models/project.rb:398]: Only select active team member whose end_date is blank

File: spec/models/company_spec.rb [hunk 2]
```diff
@@ -1,14 +1,25 @@
...
   it { should embed_many :contact_persons }
   it { should have_many :addresses }
   it { should accept_nested_attributes_for :contact_persons }
   it { should accept_nested_attributes_for :addresses }
   it { should have_many(:projects) }
   it { should validate_presence_of(:name) }
   it { should validate_uniqueness_of(:name) }
+  it { should validate_uniqueness_of(:invoice_code) }
+
+  it "Should fail as invoice code length is more than 3" do
```
Comment (Ashish944) [inline spec/models/company_spec.rb:14]: Replace to single quotes

File: app/models/project.rb (hunk 1 above)
Comment (tejaszambre) [reply app/models/project.rb:398]: Okay will update that

File: spec/models/company_spec.rb (hunk 2 above)
Comment (tejaszambre) [reply spec/models/company_spec.rb:14]: okay

File: app/models/project.rb [hunk 3]
```diff
@@ -387,7 +388,10 @@ def get_user_projects_from_project(from_date, to_date)
   private
 
   def update_user_projects
-    user_projects.each { |user_project| user_project.update_attributes(end_date: end_date) }
+    user_projects.where(active: true).each do |user_project|
+      user_project.update_attributes(end_date: end_date)
+      user_project.update_attributes(active: false) unless is_active
```
Comment (Ashish944) [inline app/models/project.rb:393]: No need for this condition check update both field at the same time

**Review by Ashish944 — APPROVED** [review]

**Comment by Sameer-Tilak** [comment]
LGTM

**Pull request merged**